s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -58,6 +60,32 @@ private[alpakka] final class S3Stream(credentials: AWSCredentials, region: Strin
   val MinChunkSize = 5242880 //in bytes
   val signingKey = SigningKey(credentials, CredentialScope(LocalDate.now(), region, "s3"))
 
+  /** listBucket lists files in a bucket, filtered by an optional prefix 
+    * 
+    * @param bucket name of the bucket
+    * @param prefix optional prefix
+    * @return A akka Source with keys
+    * 
+    */
+  def listBucket(bucket: String, prefix: Option[String] = None): Source[String, NotUsed] = {
```

Comment:
First off, before seeing this I posted https://github.com/akka/alpakka/pull/214 as the team Im on also needed this and we have forked implementation right now. A couple concerns/questions about this implementation. 

1. This is assuming that the caller wants ALL keys under a given bucket. Maybe this is okay because you could cut off the stream later if you didn't want all keys?
2. With this, if I have a very large bucket, do I start getting keys downstream while this is paging? Or does this page until the end before I see anything downstream?